### DISCUSSION TITLE
why toSql() method is not available on some methods of queryBuilder

### DISCUSSION THREAD

**Discussion by amir-khoshbakht** [laravel/framework#38750] [Q&A] [answered]
Why dd and toSql methods do not exist on all methods like (update / upsert / insert / delete)

is there any easy way to make this query with queryBuilder ?
TestTable::query()->where('id','=',1)->update(['name'=>'test name']);
UPDATE `test_table` SET `name` = 'test name' WHERE `test_table`.`id` = 1;

**Comment by dennisprudlo** [comment, marked as answer]
These methods execute the query, they do not return a query builder instance.

If these methods returned a query builder instance you would need to somehow execute the query with an additional `execute()` call or something.

You can log the queries and inspect them if you want to see which sql query was executed by Laravel.
